### PR TITLE
Update inflation

### PR DESCRIPTION
**Pull request by adamdossa**
Updates token inflation behaviour when POLYX total supply is greater than 1 billion.

# Previous behaviour

When the total supply exceeds 1 billion, a fixed amount of 140 million POLYX will be minted annually for validators / nominators.

# New behaviour

When the total supply exceeds 1 billion, and the reward curve based annual inflation (calculated based on the reward curve, staking ratio and total supply) exceeds 140 million, a fixed inflation of 140 million will be used instead.

# Rationale for change

Without this change, as soon as the total supply exceeds 1 billion, there will be a jump in the inflation rate (and corresponding interest rate for nominators) due to moving to a fixed inflation amount of 140 million annually.

Instead of this behaviour we would rather see a smooth transition, continuing to use the reward curve based mechanism until that mechanism would result in inflation greater than 140 million annually, at which point we revert to the previous logic, capping the annual inflation at 140 million.

In other words, the 140 million annual inflation becomes a cap on annual inflation, rather than a fixed value, once the total supply reaches 1 billion POLYX.

### new features

Update token inflation to ensure a smooth continuous change as the total supply crosses 1 billion. Annual inflation remains capped at 140 million after the total supply is greater than 1 billion, but can be lower if the staking ratio and curve details calculate a lower rate.

### CODE REVIEW

**Review by HenriqueNogara — APPROVED** [review]

File: pallets/staking/src/inflation.rs [hunk 1]
```diff
@@ -44,18 +44,20 @@ where
...
-    let maximum = portion * (yearly_inflation.maximum * total_tokens);
+    let maximum = portion * (yearly_inflation.maximum * total_tokens.clone());
+    // Have fixed rewards kicked in?
+    if total_tokens >= max_inflated_issuance {
+        let fixed_payout = portion * non_inflated_yearly_reward;
+        if fixed_payout <= payout {
+            // payout is always maximum.
+            return (fixed_payout.clone(), fixed_payout);
+        }
+    }
```
Comment (Neopallium) [inline pallets/staking/src/inflation.rs]: Optional minor change.  The `maximum` value can be calculated after the `if` and doesn't need the clone.
```suggestion
    // Have fixed rewards kicked in?
    if total_tokens >= max_inflated_issuance {
        let fixed_payout = portion * non_inflated_yearly_reward;
        if fixed_payout <= payout {
            // payout is always maximum.
            return (fixed_payout.clone(), fixed_payout);
        }
    }
    let maximum = portion * (yearly_inflation.maximum * total_tokens);
```

**Review by Neopallium — APPROVED** [review]
